garrysmod/lua/includes/extensions/achievements.lua

```diff
@@ -0,0 +1,52 @@
+local bitsize = 0
+local achievementsN = 0
+local achievementsLookup = {}
+
+if SERVER then
+	util.AddNetworkString("achievements")
+	achievements = {}
+
+	function achievements.Call(ply, aName, data)
+		local aID = achievementsLookup[aName]
+		if (not aID) then error("Attempt to call unregistered achievement (" .. tostring(aName) .. ")") end
+		net.Start("achievements")
+		net.WriteUInt(aID, bitsize)
+
+		if (data) then
+			net.WriteUInt(data, 16)
+		end
+
+		net.Send(ply)
+	end
+end
+
+function achievements.Register(name,func)
+	if CLIENT and not isfunction(func) then error("Second argument must be a function") end
+
+	if (SERVER) then
+		achievementsLookup[name] = achievementsN
+	else
+		achievementsLookup[achievementsN] = func
+	end
+
+	achievementsN = achievementsN + 1
+	bitsize = math.ceil(math.log(achievementsN) / math.log(2)) + 1
```

Comment:
- After `achievementsN` becomes 3, `bitsize` is larger than it should be by 1;
- `math.log(x) / math.log(base)` can be simplified to `math.log(x, base)`.

```suggestion
	bitsize = math.ceil(math.log(achievementsN + 1, 2))
```